test/zzz-rsyslog.sh

```diff
@@ -347,6 +347,8 @@ oc set env $rsyslog_ds LOGGING_FILE_PATH-
 start_rsyslog
 rpod=$( get_running_pod rsyslog )
 
+clear_and_restart_journal
```

Comment:
this doesn't really help with the new containerized test - if the problem is that the journal is so large that it takes more than 10 minutes for rsyslog to start at the beginning of the journal and read through all of the records to get to the latest one, then the best solution is to use `start_rsyslog cleanfirst` which will remove the imjournal state file and any queued records.  The test uses `RSYSLOG_JOURNAL_READ_FROM_TAIL=on RSYSLOG_FILE_READ_FROM_TAIL=on` https://github.com/openshift/origin-aggregated-logging/pull/1685/files#diff-17f5db5ab4de5254d79187c056ba91b4R154 - which means if there is no imjournal state file, and no queued records, rsyslog will start reading from the end of the journal, which means it should be extremely fast to read the very latest records

Reply:
> this doesn't really help with the new containerized test - if the problem is that the journal is so large that it takes more than 10 minutes for rsyslog to start at the beginning of the journal and read through all of the records to get to the latest one, then the best solution is to use `start_rsyslog cleanfirst` which will remove the imjournal state file and any queued records. The test uses `RSYSLOG_JOURNAL_READ_FROM_TAIL=on RSYSLOG_FILE_READ_FROM_TAIL=on` https://github.com/openshift/origin-aggregated-logging/pull/1685/files#diff-17f5db5ab4de5254d79187c056ba91b4R154 - which means if there is no imjournal state file, and no queued records, rsyslog will start reading from the end of the journal, which means it should be extremely fast to read the very latest records

Thanks!  I'll apply your suggestions.
The good news is rsyslog tests themselves have passed except hack/testing/util.sh:521: wait_for_ops_log_message...
https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_origin-aggregated-logging/1685/pull-ci-openshift-origin-aggregated-logging-master-full-integ-aws/1112/artifacts/full-integ-aws/scripts/test-zzz-rsyslog/artifacts/zzz-rsyslog.sh-artifacts.txt

Reply:
> application logs are found in journalctl output

???

Reply:
> > application logs are found in journalctl output
> 
> ???

This is the function ERROR is reported.  (exit code is 141) - [1]  I thought the problem is in journalctl.  But it seems like an bash pipe issue.  Do you think adding `trap "" PIPE` to zzz-rsyslog.sh as in [2] helps?
https://github.com/openshift/origin-aggregated-logging/blob/master/hack/testing/util.sh#L469-L475
```
[1]
[..] [ERROR] PID 130938: hack/testing/util.sh:476: `awk -v "es_ops=SYSLOG_IDENTIFIER=$uuid_es_ops" -v es_ops_out=$ARTIFACT_DIR/es_ops_out.txt '
[..] [ERROR]             BEGIN{RS="";FS="\n"};
[..] [ERROR]             $0 ~ es_ops {print > es_ops_out; exit 0}'` exited with status 141.
[..] [INFO] 		Stack Trace: 
[..] [INFO] 		  1: hack/testing/util.sh:476: `awk -v "es_ops=SYSLOG_IDENTIFIER=$uuid_es_ops" -v es_ops_out=$ARTIFACT_DIR/es_ops_out.txt '
[..] [INFO]             BEGIN{RS="";FS="\n"};
[..] [INFO]             $0 ~ es_ops {print > es_ops_out; exit 0}'`
[..] [INFO] 		  2: hack/testing/util.sh:521: wait_for_ops_log_message
[..] [INFO] 		  3: test/zzz-rsyslog.sh:383: wait_for_fluentd_to_catch_up
[..] [INFO]   Exiting with code 141.
```
```
[2]
--- a/test/zzz-rsyslog.sh
+++ b/test/zzz-rsyslog.sh
@@ -88,6 +88,7 @@ cleanup() {
     exit $return_code
 }
 trap "cleanup" EXIT
+trap "" PIPE 
```

Reply:
I'm not sure - the `[ERROR] PID 130938: hack/testing/util.sh:476: `awk -v ...` is benign - I think yes, it is related to the pipe exiting prematurely - not sure if adding `trap "" PIPE` will fix it, or if that will also mask other pipe errors which we do want to be reported.  I never could figure out how to suppress that output, I think it has something to do with the fact that it is run in the background.
At any rate, I don't think it is an error.  Check the badly[1] named `es_ops_out.txt` file in the artifacts directory to see if the record added by `logger` could be found in the journal before the test was completed.
[1] it is a bad name because it implies that it is the output from elasticsearch, when in reality it is the record that should have been added to the es ops